src/tests/test_kast.py

```diff
@@ -1,15 +1,44 @@
 from typing import Final, List, Tuple
 from unittest import TestCase
 
-from pyk.kast import KApply, KDefinition, KFlatModule, KInner, KLabel, KSequence, KSort, build_assoc
+from pyk.kast import KApply, KAtt, KDefinition, KFlatModule, KInner, KLabel, KSequence, KSort, KVariable, build_assoc
 from pyk.prelude.kbool import BOOL
 from pyk.prelude.kint import INT
 from pyk.prelude.string import STRING
 from pyk.prelude.utils import token
+from pyk.utils import FrozenDict
 
 from .utils import f, x, y, z
 
 
+class KVariableTest(TestCase):
+    TEST_DATA: Final[Tuple[Tuple[str, KInner, KInner], ...]] = (
+        ('no-sort', KVariable('Foo'), KVariable('Foo')),
+        ('sort', KVariable('Foo', sort=INT), KVariable('Foo', sort=INT)),
+        (
+            'sort-via-att',
+            KVariable('Foo', att=KAtt({'org.kframework.kore.Sort': FrozenDict(INT.to_dict())})),
+            KVariable('Foo', sort=INT),
+        ),
+        (
+            'sort-overrides-att',
+            KVariable('Foo', sort=INT, att=KAtt({'org.kframework.kore.Sort': FrozenDict(BOOL.to_dict())})),
+            KVariable('Foo', sort=INT),
+        ),
+    )
+
+    def test_construct(self) -> None:
+        for i, (name, input, expected) in enumerate(self.TEST_DATA):
+            with self.subTest(i=i, name=name):
+                self.assertEqual(input, expected)
+
+    def test_to_dict(self) -> None:
+        for i, params in enumerate(self.TEST_DATA):
+            name, input, expected = params
+            with self.subTest(i=i, name=name):
```

Comment:
```suggestion
    def test_to_dict(self) -> None:
        for i, (name, input, expected) in enumerate(self.TEST_DATA):
            with self.subTest(i=i, name=name):
```

Same change here, and anywhere else where this pattern is used in this PR.